Regression in the tests with Flask-WTF>=0.13

I manually [relaunched a test](https://travis-ci.org/MongoEngine/flask-mongoengine/jobs/156143238) to try the new Flask-WTF release and it breaks in [test_modelselectfield_multiple](https://github.com/MongoEngine/flask-mongoengine/blob/master/tests/test_forms.py#L307).

```
======================================================================
1) FAIL: test_modelselectfield_multiple (tests.test_forms.WTFormsAppTestCase)
----------------------------------------------------------------------
   Traceback (most recent call last):
    tests/test_forms.py line 307 in test_modelselectfield_multiple
      self.assertEqual(form.dogs.data, None)
   AssertionError: [<Dog: Dog object>, <Dog: Dog object>] != None
       '[<Dog: Dog object>, <Dog: Dog object>] != None' = '%s != %s' % _common_shorten_repr([<Dog: Dog object>, <Dog: Dog object>], None)
       '[<Dog: Dog object>, <Dog: Dog object>] != None' = self._formatMessage('[<Dog: Dog object>, <Dog: Dog object>] != None', '[<Dog: Dog object>, <Dog: Dog object>] != None')
   >>  raise self.failureException('[<Dog: Dog object>, <Dog: Dog object>] != None')
```

I guess something changed in Flask-WTF that breaks the test.

--- a/setup.py
+++ b/setup.py
@@ -36,7 +36,7 @@ try:
 except:
     pass
 
-test_requirements = ['coverage', 'flake8', 'mongomock', 'nose', 'rednose']
+test_requirements = ['coverage', 'mongomock', 'nose', 'rednose']
 
 setup(
     name='flask-mongoengine',

--- a/tests/test_forms.py
+++ b/tests/test_forms.py
@@ -243,7 +243,9 @@ class WTFormsAppTestCase(FlaskMongoEngineTestCase):
             class DogOwner(db.Document):
                 dog = db.ReferenceField(Dog)
 
-            DogOwnerForm = model_form(DogOwner)
+            DogOwnerForm = model_form(DogOwner, field_args={
+                'dog': { 'allow_blank': True }
+            })
 
             dog = Dog(name="fido")
             dog.save()
@@ -253,6 +255,15 @@ class WTFormsAppTestCase(FlaskMongoEngineTestCase):
 
             self.assertEqual(wtforms.widgets.Select, type(form.dog.widget))
             self.assertFalse(form.dog.widget.multiple)
+
+            # Validate the options - should contain a dog (selected) and a
+            # blank option there should be an extra blank option.
+            choices = list(form.dog)
+            self.assertEqual(len(choices), 2)
+            self.assertFalse(choices[0].checked)
+            self.assertEqual(choices[0].data, '__None')
+            self.assertTrue(choices[1].checked)
+            self.assertEqual(choices[1].data, dog.pk)
 
             # Validate selecting one item
             form = DogOwnerForm(MultiDict({
@@ -289,7 +300,6 @@ class WTFormsAppTestCase(FlaskMongoEngineTestCase):
 
             self.assertEqual(wtforms.widgets.Select, type(form.dogs.widget))
             self.assertTrue(form.dogs.widget.multiple)
-            self.assertTrue(form.dogs.widget)
 
             # Validate the options - both dogs should be selected and
             # there should be an extra blank option.

--- a/tests/test_forms.py
+++ b/tests/test_forms.py
@@ -276,7 +276,9 @@ class WTFormsAppTestCase(FlaskMongoEngineTestCase):
             class DogOwner(db.Document):
                 dogs = db.ListField(db.ReferenceField(Dog))
 
-            DogOwnerForm = model_form(DogOwner)
+            DogOwnerForm = model_form(DogOwner, field_args={
+                'dogs': { 'allow_blank': True }
+            })
 
             dogs = [Dog(name="fido"), Dog(name="rex")]
             for dog in dogs:
@@ -287,12 +289,18 @@ class WTFormsAppTestCase(FlaskMongoEngineTestCase):
 
             self.assertEqual(wtforms.widgets.Select, type(form.dogs.widget))
             self.assertTrue(form.dogs.widget.multiple)
+            self.assertTrue(form.dogs.widget)
 
-            # Validate if both dogs are selected
+            # Validate the options - both dogs should be selected and
+            # there should be an extra blank option.
             choices = list(form.dogs)
-            self.assertEqual(len(choices), 2)
-            self.assertTrue(choices[0].checked)
+            self.assertEqual(len(choices), 3)
+            self.assertFalse(choices[0].checked)
+            self.assertEqual(choices[0].data, '__None')
             self.assertTrue(choices[1].checked)
+            self.assertEqual(choices[1].data, dogs[0].pk)
+            self.assertTrue(choices[2].checked)
+            self.assertEqual(choices[2].data, dogs[1].pk)
 
             # Validate selecting two items
             form = DogOwnerForm(MultiDict({
@@ -302,7 +310,7 @@ class WTFormsAppTestCase(FlaskMongoEngineTestCase):
 
             # Validate selecting none actually empties the list
             form = DogOwnerForm(MultiDict({
-                'dogs': [],
+                'dogs': '__None',
             }), dogs=dogs)
             self.assertEqual(form.dogs.data, None)
 

